### PR TITLE
[Cache] Improve performances into foreach (PhpFilesAdapter)

### PR DESCRIPTION
**Pull request by hadjedjvincent**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| License       | MIT

Let's prevent some useless call to be made into this foreach (ini_get)

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php [hunk 1]
```diff
@@ -92,6 +92,7 @@ protected function doSave(array $values, $lifetime)
     {
         $ok = true;
         $data = array($lifetime ? time() + $lifetime : PHP_INT_MAX, '');
+        $allowCompile = ('cli' !== PHP_SAPI || ini_get('opcache.enable_cli'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: brackets not required

File: src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php [hunk 2]
```diff
@@ -116,7 +117,7 @@ protected function doSave(array $values, $lifetime)
             $file = $this->getFile($key, true);
             $ok = $this->write($file, '<?php return '.var_export($data, true).';') && $ok;
 
-            if ('cli' !== PHP_SAPI || ini_get('opcache.enable_cli')) {
+            if (true === $allowCompile) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php]: `if ($allowCompile) {`

**Comment by stof** [comment]
👍

**Comment by nicolas-grekas** [comment]
Thank you @hadjedjvincent.

**Pull request closed without merging**